File: bindgen-tests/swift/tests/external_types.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import uniffi_bindgen_tests

assert(roundtripExtRecord(rec: ExternalRec(a: 42)) == ExternalRec(a: 42))

assert(roundtripExtEnum(en: ExternalEnum.two) == ExternalEnum.two)

let interface = ExternalInterface(value: 20)
assert(roundtripExtInterface(interface: interface).getValue() == 20)

assert(roundtripExtCustomType(custom: 100) == 100);

File: bindgen-tests/python/tests/external_types.py
from test_package.uniffi_bindgen_tests import *
from test_package.uniffi_bindgen_tests_external_types_source import (
    ExternalRec, ExternalEnum, ExternalInterface)
import unittest

class TestExternalTypes(unittest.TestCase):
    def test_external_types(self):
        self.assertEqual(roundtrip_ext_record(ExternalRec(a=42)), ExternalRec(a=42))

        self.assertEqual(roundtrip_ext_enum( ExternalEnum.TWO), ExternalEnum.TWO)

        interface = ExternalInterface(20)
        self.assertEqual(roundtrip_ext_interface(interface).get_value(), 20)

        self.assertEqual(roundtrip_ext_custom_type(100), 100)

if __name__ == '__main__':
    unittest.main()
